internal/util/logging/logram.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+import (
+	"fmt"
+	"sync"
+
+	"go.uber.org/zap/zapcore"
+)
+
+// logRAM structure storage of log records in memory.
+type logRAM struct {
+	size    int64
+	counter int64
+	log     []*zapcore.Entry
+	sync.RWMutex
```

Comment:
Embedding mutex exposes its exported methods to the caller. That should be a normal field. And let's make it the first field: https://dmitri.shuralyov.com/idiomatic-go/entries/2